src/shader_recompiler/ir/passes/srt.h

```diff
@@ -0,0 +1,126 @@
+// SPDX-FileCopyrightText: Copyright 2024 shadPS4 Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include <vector>
+#include <boost/align/aligned_allocator.hpp>
+#include <boost/align/aligned_delete.hpp>
+#include <boost/container/map.hpp>
+#include <boost/container/set.hpp>
+#include <boost/container/small_vector.hpp>
+
+#include <memory>
+#include "common/alignment.h"
+#include "common/assert.h"
+#include "common/types.h"
+#include "xbyak/xbyak.h"
+
+namespace Shader {
+
+class FlattenedUserDataBuffer {
```

Comment:
This abstraction probably is superfluous and can be a std::vector<u32> directly in info. Or a type alias ```using FlattenedUserDataBuffer = std::vector<u32>;``` if it makes sense